src/__experimental__/components/date/date.component.js

```diff
@@ -132,7 +137,14 @@ class BaseDateInput extends React.Component {
     const visibleValue = formatDateToCurrentLocale(date);
     const newDate = this.getDateObject(date);
 
-    this.setState({ selectedDate: newDate, visibleValue },
+    this.setState({
+      selectedDate: newDate,
+      visibleValue,
+      lastValidEventValues: {
+        formattedValue: visibleValue,
+        rawValue: DateHelper.formatValue(visibleValue)
```

Comment:
There's a case of unfortunate naming here.

You'd assume that the return value of `DateHelper.formatValue()` should be assigned to `formattedValue`, so it looks like a mistake that it's being assigned to `rawValue` instead.

But I realise that `DateHelper.formatValue()` returns an ISO date string (`YYYY-MM-DD`), which is what we want in `rawValue`.

It'd be good if `DateHelper` had a method called `isoValue()` for use in situations like this.

Reply:
Yeah I agree, I don't really want to rename it in the helper just in case end users have actually used it in the consumer apps, I will look to make it more readable in my implementation

Reply:
Thanks for refactoring with that new `isoFormattedValueString()` function :thumbsup: